### PR TITLE
added onion protocol

### PR DESCRIPTION
**Pull request by ricott1**
Migrated pull from https://github.com/ensdomains/ens-app/pull/355

I added the option to register an onion address, encoded as a [multiaddress](https://github.com/multiformats/js-multiaddr), in the `contenthash` entry.

This pull will work in tandem with a [metamask pull](https://github.com/MetaMask/metamask-extension/pull/6629).

It depends on another pull made on the [content-hash](https://github.com/pldespaigne/content-hash/pull/24) dependency, which we should wait for.

### CODE REVIEW

**Comment by jefflau** [comment]
@ricott1 Thanks for this - please ping me when that pull request has gone through

**Comment by jefflau** [comment]
@ricott1 Also can you please add a test for the new protocol

**Comment by pldespaigne** [comment]
Hi @jefflau ! I just introduce a general encoding function for content-hash in [this PR](), it is published to npm as v2.4.1.
It is now possible to do `contentHash.encode('onion', onionAddress)`, I will update the readme's api section later today.

File: src/utils/contents.js [hunk 1]
```diff
@@ -50,6 +54,8 @@ export function encodeContenthash(text) {
         encoded = '0x' + contentHash.fromIpfs(content)
       } else if (contentType === 'bzz') {
         encoded = '0x' + contentHash.fromSwarm(content)
+      } else if (contentType === 'onion') {//In the future insert here general encoding given by contentHash library
+        encoded = '0x' + contentHash.fromOnion(content)
```
Comment (pldespaigne) [inline src/utils/contents.js]: this should be `contentHash.encode('onion', content);`

Comment (ricott1) [reply src/utils/contents.js]: done

**Comment by ricott1** [comment]
Tests added

**Comment by pldespaigne** [comment]
:tada:

**Pull request merged**